### PR TITLE
Add packages OSPRay, rkcommon, Open VKL, and Open Image Denoise

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Add OSPRay and additional required packages to Spack.

Part 1 in addressing #32656 .

@mvictoras @vicentebolea

### CODE REVIEW

File: var/spack/repos/builtin/packages/openimagedenoise/package.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    version("1.4.3", sha256="37941f260af212183579efb6b1e07e4075e3ab9cbd63120afd007e62367efef6")
+    version("1.4.2", sha256="742de28e2c018dc47819aaeac039e504df1ce660ed72aa0b49009343c0a9aae1")
+    version("1.4.1", sha256="db022565ffe46266188992e1db68d11c94f799c37e25996f5aee44bebffcf079")
+    version("1.4.0", sha256="4f285b0f58d3aa4419afd1bf57784416087119dc620dcf1a6b9fa19a27b32c5b")
+    version("1.3.0", sha256="a6b30d7aa3ea81490431dba823dab475daad73a7124940560624742126db7274")
+    version("1.2.4", sha256="f847c244417e68ba46149663055f163188cc2ec62308fe040ec943df184a2c40")
+    version("1.2.3", sha256="9ff7f7b5c442bd56be1ae89f8c45ae1972498b854f7175705ff520bb2c5cb25b")
+
+    depends_on("ispc")
+    depends_on("python@3:")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/openimagedenoise/package.py]: Build and link or just build?

Comment (kwryankrattiger) [reply var/spack/repos/builtin/packages/openimagedenoise/package.py]: good catch, it is build and test only. ispc is also only a build dep fixed that too.

**Comment by kwryankrattiger** [comment]
Tested this with `ospray@2.7.0,2.7.1,2.9.0`, `rkcommon@1.9.0`, `openvkl@1.2.0`, and `openimagedenoise@1.4.3`.

`ospray@2.10..0` and `rkcommon@1.10.0` both seem to require additional investigation to build successfully. They are implicitly disabled via disabling `ispc@1.17:`, which currently has a link issue with `llvm` and `ncurses` tinfo.

**Comment by kwryankrattiger** [comment]
```
>> 271    ld: /build/src/opt/spack/linux-ubuntu20.04-zen2/gcc-11.1.0/llvm-12.0.1-p7l5yna2otpq6gfy2elpzuliodaj4idq/lib/libLLVMSupport.a(Process.cpp.o): undefined reference to symbol 'del_curterm@@@@NCURSES6_TINFO_5.0.19991023'
>> 272    ld: /build/src/opt/spack/linux-ubuntu20.04-zen2/gcc-11.1.0/ncurses-6.4-3jf6viqr2dkimusd62yjwpsns7assj6k/lib/libtinfo.so.6: error adding symbols: DSO missing from command line
>> 273    clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
>> 274    make[2]: *** [CMakeFiles/ispc.dir/build.make:1435: bin/ispc] Error 1
   275    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-ispc-1.18.0-saosdj3npab7clcikxqnagaoti4yttj2/spack-build-saosdj3'
>> 276    make[1]: *** [CMakeFiles/Makefile2:186: CMakeFiles/ispc.dir/all] Error 2
   277    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-ispc-1.18.0-saosdj3npab7clcikxqnagaoti4yttj2/spack-build-saosdj3'
>> 278    make: *** [Makefile:139: all] Error 2
```

**Comment by kwryankrattiger** [comment]
@haampie This should be ready to merge now, the pieces required for ParaView with OSPRay rendering are there which is what this is really for.

**Review by haampie — APPROVED** [review]

**Pull request merged**